### PR TITLE
Added support for secondary keybindings

### PR DESCRIPTION
**Pull request by jefetienne**
[[Images](https://imgur.com/a/7wrMblw)]

Adds support for secondary keybinds, which can be accessed via a small toggle button at the top right of the controls window. By default, they are unpopulated, although later in the future I will create ControllerConfigs that will have loadout configurations specific to the controller type and OS.

Any cross-unary conflicts will be highlighted as blue (to help colorblind people distinguish it from yellow or red/"dark yellow"), and the player cannot "Continue" until resolving them, but can switch between primary and secondary. Any internal (red) conflicts made, and they cannot switch between primary and secondary until that gets resolved first.

To not bloat the controls window and avoid static code, I made a ControlsConfigManager that sets aside much of the cross-window logic from the main controls window. Also fixed a bug with unknown actions - if the player had an unknown action that was bound to a certain key, using that same key to rebind to an existent action would cause an exception and fail to update Keybinds.txt. Now, if it is rebinded, it will erase the unknown action (as it should be repopulated anyway if they switch to a newer version.)

### CODE REVIEW

**Comment by jefetienne** [comment]
I also think it would be good to remove bindings to "None" by right-clicking a keybind button and prompting "are you sure you want to clear the binding for ActionName ('ButtonName')?", but I will add that as a separate PR after this.

**Comment by jotoho** [comment]
> I also think it would be good to remove bindings to "None" by right-clicking a keybind button and prompting "are you sure you want to clear the binding for ActionName ('ButtonName')?", but I will add that as a separate PR after this.

I wanted to suggest allowing players to unbind keys they do not want by "assigning" it to the Escape key, like in many other games, to make it more likely that players will find it, but then noticed that the pause menu can be reassigned through the advanced controls menu, so that's not really an option, is it?

Being able to unbind keys at all would be a useful improvement but I feel like right-clicking is not necessarily obvious enough, so I don't know how many players would know it is possible. On the other hand, I do not know how it could be made any more easy to find, besides outright saying it in the menu.

Curiosity: Was key unbinding a thing in the original engine? I never played it for more than 5 minutes.

**Comment by jefetienne** [comment]
> I wanted to suggest allowing players to unbind keys they do not want by "assigning" it to the Escape key, like in many other games, to make it more likely that players will find it, but then noticed that the pause menu can be reassigned through the advanced controls menu, so that's not really an option, is it?

Typically I think the Escape key is more for canceling than removing a keybind, but yes that is a reason. Maybe in the future I'll set Escape back to a reserved keybind but treat the Pause keybind as an exception. I have seen games though where right-clicking does ask to remove keybinds.

File: Assets/Scripts/Game/UserInterfaceWindows/DaggerfallControlsWindow.cs [hunk 1]
```diff
@@ -294,6 +247,8 @@ private void UpdateKeybindButtons()
             SetupKeybindButtons(lookKeys, 32, 36, 269, 102, false);
             SetupKeybindButtons(uiKeys, 36, 40, 269, 147, false);
 
+            currentBindingsButton.Label.Text = ControlsConfigManager.Instance.UsingPrimary ? "Primary" : "Secondary";
```
Comment (Interkarma) [inline Assets/Scripts/Game/UserInterfaceWindows/DaggerfallControlsWindow.cs:250]: Don't let me forget to add these strings to the Internal_Strings collection for localization later. :)

**Comment by Interkarma** [comment]
As always, this just works perfectly @jefetienne. No problems found on this end, I think it's well worth getting into 0.10.26.

File: Assets/Scripts/Game/UserInterfaceWindows/DaggerfallControlsWindow.cs (hunk 1 above)
Comment (jefetienne) [reply Assets/Scripts/Game/UserInterfaceWindows/DaggerfallControlsWindow.cs:250]: Got it :)

**Pull request merged**